sql/size_utils.sql

```diff
@@ -682,3 +635,126 @@ BEGIN
+        _timescaledb_catalog.chunk_index ci
+        WHERE ch.schema_name = n.nspname
+            AND c.relnamespace = n.oid
+            AND c.relname = ci.index_name
+            AND ch.id = ci.chunk_id
+            AND h.id = ci.hypertable_id
+            AND ci.hypertable_index_name = index_name_in
+        GROUP BY ci.hypertable_id; 
+END;
+$BODY$;
+
+CREATE OR REPLACE FUNCTION _timescaledb_internal.data_node_index_size (node_name name, index_name_in name)
+RETURNS TABLE ( hypertable_id INTEGER, total_bytes BIGINT)
+AS '@MODULE_PATHNAME@' , 'ts_dist_remote_hypertable_index_info' LANGUAGE C VOLATILE STRICT;
+
+CREATE OR REPLACE FUNCTION _timescaledb_internal.indexes_remote_size(
+    schema_name_in             NAME,
+    table_name_in              NAME,
+    index_name_in              NAME
+)
+RETURNS TABLE (
+    total_bytes bigint,
+    node_name   NAME)
+               LANGUAGE PLPGSQL STABLE STRICT
+               AS
+$BODY$
+BEGIN
+    RETURN QUERY
+    SELECT
+        sum(entry.total_bytes)::bigint AS total_bytes,
+        srv.node_name
+    FROM (
+        SELECT
+            s.node_name,
+            _timescaledb_internal.ping_data_node (s.node_name) AS node_up
+        FROM
+            _timescaledb_catalog.hypertable AS ht,
+            _timescaledb_catalog.hypertable_data_node AS s
+        WHERE
+            ht.schema_name = schema_name_in
+            AND ht.table_name = table_name_in
+            AND s.hypertable_id = ht.id
+         ) AS srv
+    LEFT OUTER JOIN LATERAL _timescaledb_internal.data_node_index_size(
+    CASE WHEN srv.node_up THEN
+        srv.node_name
+    ELSE
+        NULL
+    END, index_name_in) entry ON TRUE
+    GROUP BY srv.node_name;
+END;
+$BODY$;
+
+-- Get sizes of indexes on a hypertable
+--
+-- index_name           - index on hyper table
+--
+-- Returns:
+-- total_bytes          - size of index on disk
+
+CREATE OR REPLACE FUNCTION  hypertable_index_relation_size(
+    index_name              REGCLASS
+)
+RETURNS TABLE (
+               total_bytes BIGINT,
+               node_name   NAME)
+LANGUAGE PLPGSQL STABLE STRICT
+AS
+$BODY$
+<<main>>
+DECLARE
+        ht_index_name       NAME;
+        ht_schema_name      NAME;
+        ht_name      NAME;
+        is_distributed   BOOL;
+        ht_id INTEGER;
+BEGIN
+
+   --select c.relname, cind.indexrelid, cind.indrelid, cl.relname, nsp.nspname       
+   SELECT c.relname, cl.relname, nsp.nspname       
+   INTO STRICT ht_index_name, ht_name, ht_schema_name  
+   FROM pg_class c, pg_index cind, pg_class cl, pg_namespace nsp
+   WHERE c.oid = cind.indexrelid and cind.indrelid = cl.oid
+         and cl.relnamespace = nsp.oid and c.oid = index_name;
+        
+   SELECT replication_factor > 0
+   INTO STRICT is_distributed
+   FROM _timescaledb_catalog.hypertable ht
+   WHERE ht.schema_name = ht_schema_name and ht.table_name = ht_name;
+
+   CASE WHEN is_distributed THEN
+         --RETURN QUERY SELECT NULL, NULL ; 
+         RETURN QUERY SELECT *
```

Comment:
It will be more clear:
```suggestion
         RETURN QUERY SELECT ch.total_bytes, ch.node_name
```